src/main/kotlin/uk/gov/dluhc/emailnotifications/SesEmailClient.kt

```diff
@@ -22,6 +22,7 @@ class SesEmailClient(
         val filteredEmailToAddresses = filterRecipientsIfAllowListConfigured(emailToRecipients, subject)
         val filteredEmailCcAddresses = filterRecipientsIfAllowListConfigured(emailCcRecipients, subject)
         if (filteredEmailToAddresses.isEmpty() && filteredEmailCcAddresses.isEmpty()) {
+            logger.warn { "Both toAddresses and ccAddresses have been filtered down to empty sets. No recipients to send to." }
             return emptySet()
```

Comment:
I do wonder why this returns "successfully", rather than throwing an exception (which the other fail cases do .... but I'm not going to change that)

Reply:
I think it's for a scenario If the recipients are not from `allowedList` then after filtering to/cc, these could be empty, but I agree with you. I remember I raised too in PR that this could return an exception but don't remember the real reason why it just returns an empty set